.github/workflows/backend-dev-server.yml

```
@@ -41,55 +41,38 @@ jobs:
       - name: Run Unit Tests
         run: ./gradlew test
 
-      - name: Upload Jar
-        uses: actions/upload-artifact@v4
+      - name: Download OpenTelemetry Agent
+        run: curl -L -o backend/opentelemetry-javaagent.jar https://github.com/open-telemetry/opentelemetry-java-instrumentation/releases/latest/download/opentelemetry-javaagent.jar
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v2
```

actionlint (1.7.8)

the runner of "docker/login-action@v2" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)